realm/realm-library/src/main/java/io/realm/DynamicRealmObject.java

```diff
@@ -683,35 +683,35 @@ public String toString() {
         if (row == null || !row.isAttached()) {
             return "Invalid object";
         }
-        StringBuilder sb = new StringBuilder(row.getTable().getName() + " = [");
+
+        final String className = Table.tableNameToClassName(row.getTable().getName());
+        StringBuilder sb = new StringBuilder(className + " = [");
         String[] fields = getFieldNames();
         for (String field : fields) {
             long columnIndex = row.getColumnIndex(field);
             RealmFieldType type = row.getColumnType(columnIndex);
             sb.append("{");
             switch (type) {
-                case BOOLEAN: sb.append(field).append(": ").append(row.getBoolean(columnIndex)); break;
-                case INTEGER: sb.append(field).append(": ").append(row.getLong(columnIndex)); break;
-                case FLOAT: sb.append(field).append(": ").append(row.getFloat(columnIndex)); break;
-                case DOUBLE: sb.append(field).append(": ").append(row.getDouble(columnIndex)); break;
-                case STRING: sb.append(field).append(": ").append(row.getString(columnIndex)); break;
-                case BINARY: sb.append(field).append(": ").append(Arrays.toString(row.getBinaryByteArray(columnIndex))); break;
-                case DATE: sb.append(field).append(": ").append(row.getDate(columnIndex)); break;
+                case BOOLEAN: sb.append(field).append(":").append(row.isNull(columnIndex) ? "null" : row.getBoolean(columnIndex)); break;
```

Comment:
I think it is too long for one line... use the normal format of `switch ... case` block?